File: backend/src/ShaktiUdyog.Domain/Constants/Roles.cs
namespace ShaktiUdyog.Domain.Constants;

/// <summary>
/// Application role names. These are seeded at startup and referenced by
/// authorization policies. Keep in sync with docs/shakti-udyog-requirements.md §13.
/// </summary>
public static class Roles
{
    public const string Admin = "Admin";
    public const string Engineer = "Engineer";
    public const string Customer = "Customer";

    public static readonly IReadOnlyList<string> All = [Admin, Engineer, Customer];
}

File: frontend/src/auth/roles.ts
/**
 * Application role names. Must match the backend role constants
 * (backend/src/ShaktiUdyog.Domain/Constants/Roles.cs). These are used for
 * role-based routing only — real authorization is always enforced by the API.
 */
export const Roles = {
  Admin: "Admin",
  DataUpdater: "DataUpdater",
  Customer: "Customer",
} as const;

export type Role = (typeof Roles)[keyof typeof Roles];
